logger/common.go

```diff
@@ -117,34 +116,35 @@ func NewInfo(containerID string, containerName string, options ...InfoOpt) *dock
 
 // Start starts the actual logger.
 func (l *Logger) Start(uid int, gid int, cleanupTime *time.Duration, ready func() error) error {
-	var wg sync.WaitGroup
+	errChan := make(chan error)
 	if l.Stdout != nil {
-		wg.Add(1)
-		go l.sendLogs(l.Stdout, &wg, sourceSTDOUT, uid, gid, cleanupTime)
+		go l.sendLogs(errChan, l.Stdout, sourceSTDOUT, uid, gid, cleanupTime)
 	}
 	if l.Stderr != nil {
-		wg.Add(1)
-		go l.sendLogs(l.Stderr, &wg, sourceSTDERR, uid, gid, cleanupTime)
+		go l.sendLogs(errChan, l.Stderr, sourceSTDERR, uid, gid, cleanupTime)
 	}
 
 	// Signal that the container is ready to be started
 	if err := ready(); err != nil {
 		return errors.Wrap(err, "failed to check container ready status")
 	}
-	wg.Wait()
+
+	err := <- errChan
+	if err != nil {
+		return err
+	}
 
 	return nil
 }
 
 // sendLogs sends logs to destination.
-func (l *Logger) sendLogs(f io.Reader, wg *sync.WaitGroup, source string, uid int, gid int, cleanupTime *time.Duration) {
-	defer wg.Done()
-
+func (l *Logger) sendLogs(errChan chan error, f io.Reader, source string, uid int, gid int, cleanupTime *time.Duration) {
```

Comment:
Do we need to send a `nil` into `errChan` in the success case? Otherwise it seems like L132 in this file will block indefinitely.

Reply:
Good point, we do not send `nil` to the channel. Based on my manual testing results, I did not observe errors in the success case. But I think it's better to send a `nil` message to `errChan` for safety